pxr/usd/plugin/usdAbc/alembicReader.cpp

```diff
@@ -3086,11 +3134,25 @@ _ReadPolyMesh(_PrimReaderContext* context)
         SdfValueTypeNames->IntArray,
         _CopyGeneric<IInt32ArrayProperty, int>(
             context->ExtractSchema(".faceCounts")));
-    context->AddProperty(
-        _GetUVPropertyName(),
-        _GetUVTypeName(),
-        _CopyGeneric<IV2fGeomParam, GfVec2f>(
-            context->ExtractSchema("uv")));
+
+    // special case for texture coordinates: we want to preserve the topology,
+    // so we need to extract the indices properties if they exist.
+    auto uvProperty = context->ExtractSchema("uv");
+    if (uvProperty.Cast<IV2fGeomParam>().isIndexed()) {
```

Comment:
I'm not super familiar with the Alembic API, but if there's any chance that Cast() can fail, maybe test the result before dereferncing?